Source/SwiftLintBuiltInRules/Rules/Lint/UnhandledTryInTaskRule.swift

```diff
@@ -0,0 +1,170 @@
+        description: "Errors thrown inside of this task are not handled",
+        kind: .lint,
+        nonTriggeringExamples: [
+            Example("""
+            Task<Void, Never> {
+              try await myThrowingFunction()
+            }
+            """),
+            Example("""
+            Task {
+              try? await myThrowingFunction()
+            }
+            """),
+            Example("""
+            Task {
+              try! await myThrowingFunction()
+            }
+            """),
+            Example("""
+            Task<Void, String> {
+              let text = try myThrowingFunction()
+              return text
+            }
+            """),
+            Example("""
+            Task {
+              do {
+                try myThrowingFunction()
+              } catch let e {
+                print(e)
+              }
+            }
+            """),
+            Example("""
+            func someFunction() throws {
+              Task {
+                anotherFunction()
+                do {
+                  try myThrowingFunction()
+                } catch {
+                  print(error)
+                }
+              }
+
+              try something()
+            }
+            """),
+            Example("""
+            let task = Task {
+              try await myThrowingFunction()
+            }
+            """),
+            Example("""
+            var task = Task {
+              try await myThrowingFunction()
+            }
+            """)
+        ],
+        triggeringExamples: [
+            Example("""
+            ↓Task {
+              try await myThrowingFunction()
+            }
+            """),
+            Example("""
+            ↓Task {
+              let text = try myThrowingFunction()
+              return text
+            }
+            """),
+            Example("""
+            ↓Task {
+              do {
+                try myThrowingFunction()
+              }
+            }
+            """),
+            Example("""
+            ↓Task {
+              do {
+                try myThrowingFunction()
+              } catch let e as FooError {
+                print(e)
+              }
+            }
+            """),
+            Example("""
+            ↓Task {
+              do {
+                throw FooError.bar
+              }
+            }
+            """),
+            Example("""
+            ↓Task {
+              throw FooError.bar
+            }
+            """)
+        ]
+    )
+
+    func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
+        Visitor(viewMode: .sourceAccurate)
+    }
+}
+
+private extension UnhandledTryInTaskRule {
+    final class Visitor: ViolationsSyntaxVisitor {
+        override func visitPost(_ node: FunctionCallExprSyntax) {
+            if let typeIdentifier = node.calledExpression.as(IdentifierExprSyntax.self) {
+                if typeIdentifier.identifier.text == "Task" {
```

Comment:
Should we also check to make sure that placeholder generics aren't used? E.g. `Task<_, _>`